js/oop/Factorial.js

```diff
@@ -11,7 +11,23 @@ export class Factorial {
     this.n = parseNumber(n);
   }
 
-  // write your code here
+  countUsingLoop() {
+    let result = 1;
+
+    for (let index = this.n; index > 0; index--) {
+      result = result * index;
+    }
+
+    return result;
+  }
+
+  countUsingRecursive() {
+    if (this.n < 2) {
+      return 1;
+    }
+
+    return this.n * new Factorial(this.n - 1).countUsingRecursive();
```

Comment:
No need to re-instance the Factorial class. Just call the countUsingRecursive() method using "this" statement.